src/datasources/results/ResultsConfigEditor.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { ChangeEvent, useCallback } from 'react';
+import { DataSourcePluginOptionsEditorProps } from '@grafana/data';
+import { DataSourceHttpSettings, InlineField, InlineSegmentGroup, InlineSwitch, Text } from '@grafana/ui';
+import { ResultsDataSourceOptions, ResultsFeatureTogglesDefaults } from './types/types';
+
+interface Props extends DataSourcePluginOptionsEditorProps<ResultsDataSourceOptions> {}
+
+export const ResultsConfigEditor: React.FC<Props> = ({ options, onOptionsChange }) => {
+    const handleFeatureChange = useCallback(
+    (featureKey: string) => (event: ChangeEvent<HTMLInputElement>) => {
+      const jsonData = {
+        ...options.jsonData,
+        ...{ featureToggles: { ...options.jsonData.featureToggles, [featureKey]: event.target.checked } },
```

Comment:
```suggestion
        featureToggles: { ...options.jsonData.featureToggles, [featureKey]: event.target.checked },
```

Why do we need a copy here?

Reply:
The spread operator is not necessary because we are already creating a new featureToggles object with `{ ...options.jsonData.featureToggles, [featureKey]: event.target.checked }`